From cc4bc22447d15deacf44bc1d369621a63c934427 Mon Sep 17 00:00:00 2001
From: "Jason A. Donenfeld" <Jason@zx2c4.com>
Date: Fri, 25 Jul 2014 02:02:48 +0200
Subject: [PATCH] osflags: use pkg-config for systemd support

Recent versions of systemd don't ship with libsystemd-daemon.so anymore,
but instead use libsystemd.so for everything. This is obviously
problematic for using the same LDFLAGS on old systemd and new systemd.
So, they also ship compatability pkgconfig files, which use the old
names but return the new library. So, the most portable way to support
both old and new systemd is to use pkgconfig. It's not a problem either,
since systems that use systemd are bound to also have pkgconfig
installed.
---
 src/osflags | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/osflags b/src/osflags
index 9eda8f0..0f8a26c 100755
--- a/src/osflags
+++ b/src/osflags
@@ -19,7 +19,7 @@ link)
 		Linux)
 			FLAGS="";
 			[ -e /usr/include/selinux/selinux.h ] && FLAGS="$FLAGS -lselinux";
-			[ -e /usr/include/systemd/sd-daemon.h ] && FLAGS="$FLAGS -lsystemd-daemon";
+			[ -e /usr/include/systemd/sd-daemon.h ] && FLAGS="$FLAGS $(pkg-config --libs libsystemd-daemon)";
 			echo $FLAGS;
 		;;
 	esac
-- 
1.9.3

